File: legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/android/DataCryptUtils.java
package com.xiaomi.channel.commonutils.android;

import java.security.InvalidAlgorithmParameterException;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import javax.crypto.BadPaddingException;
import javax.crypto.Cipher;
import javax.crypto.IllegalBlockSizeException;
import javax.crypto.NoSuchPaddingException;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/android/DataCryptUtils.class */
public class DataCryptUtils {
    private static final byte[] DEFAULT_IV = {100, 23, 84, 114, 72, 0, 4, 97, 73, 97, 2, 52, 84, 102, 18, 32};

    private static Cipher createCipher(byte[] bArr, int i) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeyException, InvalidAlgorithmParameterException {
        SecretKeySpec secretKeySpec = new SecretKeySpec(bArr, "AES");
        IvParameterSpec ivParameterSpec = new IvParameterSpec(DEFAULT_IV);
        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
        cipher.init(i, secretKeySpec, ivParameterSpec);
        return cipher;
    }

    public static byte[] mipushDecrypt(byte[] bArr, byte[] bArr2) throws BadPaddingException, NoSuchPaddingException, IllegalBlockSizeException, NoSuchAlgorithmException, InvalidKeyException, InvalidAlgorithmParameterException {
        return createCipher(bArr, 2).doFinal(bArr2);
    }

    public static byte[] mipushEncrypt(byte[] bArr, byte[] bArr2) throws Exception {
        return createCipher(bArr, 1).doFinal(bArr2);
    }
}

File: legacy/src/main/java/com/xiaomi/channel/commonutils/android/DataCryptUtils.kt
package com.xiaomi.channel.commonutils.android

import javax.crypto.Cipher
import javax.crypto.spec.IvParameterSpec
import javax.crypto.spec.SecretKeySpec

/*
 * Current override reference: com.xiaomi.xmsf 0.3.17-20260410000745 (versionCode 1003003000),
 * base.apk sha256 f3d72b6f5e1427ceecd3147a051d58e4dc95bb528397d486658e01cad9f7e590,
 * JADX path: com.xiaomi.xmsf/current/base/sources/com/xiaomi/channel/commonutils/android/DataCryptUtils.java
 */
object DataCryptUtils {
    private val DEFAULT_IV = byteArrayOf(100, 23, 84, 114, 72, 0, 4, 97, 73, 97, 2, 52, 84, 102, 18, 32)

    private fun createCipher(key: ByteArray, mode: Int): Cipher {
        val secretKeySpec = SecretKeySpec(key, "AES")
        val ivParameterSpec = IvParameterSpec(DEFAULT_IV)
        return Cipher.getInstance("AES/CBC/PKCS5Padding").apply {
            init(mode, secretKeySpec, ivParameterSpec)
        }
    }

    @JvmStatic
    @Throws(Exception::class)
    fun mipushDecrypt(key: ByteArray, payload: ByteArray): ByteArray {
        return createCipher(key, Cipher.DECRYPT_MODE).doFinal(payload)
    }

    @JvmStatic
    @Throws(Exception::class)
    fun mipushEncrypt(key: ByteArray, payload: ByteArray): ByteArray {
        return createCipher(key, Cipher.ENCRYPT_MODE).doFinal(payload)
    }
}
